### PR TITLE
MueLu: Adding Teko/MueLu driver

### PR DESCRIPTION
**Pull request by csiefer2**
Adding a driver to make sure we keep the code compiling

### CODE REVIEW

File: packages/muelu/example/basic/Teko.cpp [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+// Teuchos includes /*@ \label{lned:being-includes} @*/
+#include "Teuchos_ConfigDefs.hpp"
+#include "Teuchos_GlobalMPISession.hpp"
+#include "Teuchos_RCP.hpp"
+#include "Teuchos_XMLParameterListHelpers.hpp"
+#include "Teuchos_DefaultComm.hpp"
+#include "Teuchos_AbstractFactoryStd.hpp"
+#include "Teuchos_CommandLineProcessor.hpp"
+
+// Epetra includes
```
Comment (cgcgcg) [inline packages/muelu/example/basic/Teko.cpp:56]: Remove the Epetra includes?

File: packages/muelu/example/basic/Teko.cpp [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+#include "Teko_SIMPLEPreconditionerFactory.hpp"
+#include "Teko_TpetraInverseFactoryOperator.hpp"
+
+#include "Thyra_TpetraLinearOp.hpp"
+#include "Thyra_TpetraVectorSpace.hpp"
+
+#include "Stratimikos_DefaultLinearSolverBuilder.hpp"
+#include "Stratimikos_MueLuHelpers.hpp"
+
+#include "Thyra_Ifpack2PreconditionerFactory.hpp"
```
Comment (cgcgcg) [inline packages/muelu/example/basic/Teko.cpp:89]: Might need to check if Ifpack2 is enabled.

File: packages/muelu/example/basic/Teko.cpp (hunk 1 above)
Comment (csiefer2) [reply packages/muelu/example/basic/Teko.cpp:56]: Done

File: packages/muelu/example/basic/Teko.cpp (hunk 2 above)
Comment (csiefer2) [reply packages/muelu/example/basic/Teko.cpp:89]: I just added that to the CmakeLists.txt instead.

File: packages/muelu/example/basic/Teko.cpp [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+#include "Teuchos_RCP.hpp"
+#include "Teuchos_XMLParameterListHelpers.hpp"
+#include "Teuchos_DefaultComm.hpp"
+#include "Teuchos_AbstractFactoryStd.hpp"
+#include "Teuchos_CommandLineProcessor.hpp"
+
+// Epetra includes
+#include "mpi.h"
+
+#include "Epetra_MpiComm.h"
```
Comment (cgcgcg) [inline packages/muelu/example/basic/Teko.cpp:59]: How about the other ones?

Comment (cgcgcg) [reply packages/muelu/example/basic/Teko.cpp:59]: Oh, did you forget to push?

Comment (csiefer2) [reply packages/muelu/example/basic/Teko.cpp:59]: Much like Pokemon, I'll catch them all.  I just want to make sure it compiles first :)

**Review by cgcgcg — DISMISSED** [review]

**Review by cgcgcg — APPROVED** [review]

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**